builder/vsphere/driver/datastore_test.go

```diff
@@ -35,20 +35,39 @@ func TestDatastoreIsoPath(t *testing.T) {
 		{
 			isoPath: "[datastore][] /dir/subdir/file",
 			valid:   false,
+		},{
+			isoPath: "[data/store] /dir/subdir/file",
+			valid:   false,
+		},
+		{
+			isoPath:  "   [datastore] /dir/subdir/file",
+			filePath: "/dir/subdir/file",
+			valid:    true,
+		},
```

Comment:
Do we ever care about the 'datastore' string ? Is it always datastore ? Can it be something more complicated ? Do we ever need to extract it ?

Reply:
Actually the error was happening in the datastore string because I was expecting `\w*` which means zero or more any word type character but actually I had to exclude the non-valid character which is `[`, `]` and `/`. 
We don't need to extract it here, only the file path.

Reply:
I added some tests above with the invalid `/` character and a white space like you suggested in the datastore name

Reply:
nice !

Reply:
No, it's not always [datastore1] or similar. We have 6 different datastores on our vSphere.

Reply:
The problem was specifically with datastore names containing non-word characters. We are aware of different datastore names, just the regex wasn't compatible with that.